### PR TITLE
Add an architecture chart to display project and platform specificity

### PR DESCRIPTION
**Pull request by samparent97**
Reasons for this change:
- Everytime I explain the architecture to someone, this plot is the single thing that helps it click the most, so I thought it belonged at the top of the architecture section of our docs!

Notes:
- This only renders white in dark mode 😢 
- The specific language we use may have changed
- Maybe change the surrounding descriptions to cater more to the plot?

### CODE REVIEW

**Comment by BlakeFreer** [comment]
I'll update the graphic to make it a bit tighter. I don't like it at the top of the architecture page. It should go at top the "Implementation" section, because a new reader does not have any context when they see this page for the first time.

**Comment by samparent97** [comment]
> I'll update the graphic to make it a bit tighter. I don't like it at the top of the architecture page. It should go at top the "Implementation" section, because a new reader does not have any context when they see this page for the first time.

I don't see that section, but I think it belongs somewhere where it will get views from new contributors. I have done this drawing now for both Hydar and Josh and this is what they said really made the whole thing click.

**Comment by BlakeFreer** [comment]
> > I'll update the graphic to make it a bit tighter. I don't like it at the top of the architecture page. It should go at top the "Implementation" section, because a new reader does not have any context when they see this page for the first time.
> 
> I don't see that section, but I think it belongs somewhere where it will get views from new contributors. I have done this drawing now for both Hydar and Josh and this is what they said really made the whole thing click.

The implementation section is the 3rd heading on the architecture page. I agree with including the image but not before the reader even knows that we have a distinction between platform-specific and -agnostic

**Comment by samparent97** [comment]
@TylerStAmour can you give this one a review since Blake and I have both contributed

File: docs/docs/firmware/architecture/index.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 # Architecture
 
-Our firmware is separated in 2 layers: application and platform.
+Code in our monorepo is organized based on two binary traits: Project (ECU) Specificity and Platform Specificity.
+
+![Architecture Organization](arch_diagram-Light.png#only-light){width=100%}
+![Architecture Organization](arch_diagram-Dark.png#only-dark){width=100%}
```
[the remark this answers is not in the document]
Comment (BlakeFreer) [reply docs/docs/firmware/architecture/index.md:6]: Yeah it's an mkdocs thing, not standard markdown

**Review by BlakeFreer — APPROVED** [review]

**Review by TylerStAmour — APPROVED** [review]

**Pull request merged**